prov/tcp/src/tcpx_ep.c

```diff
@@ -253,20 +239,14 @@ static int tcpx_pep_sock_create(struct tcpx_pep *pep)
 			strerror(ofi_sockerr()));
 		return -FI_EIO;
 	}
-
+	ret = tcpx_setup_socket(pep->sock);
+	if (ret) {
+		goto err;
+	}
 	if (ofi_addr_get_port(pep->info->src_addr) != 0 || port_range.high == 0) {
-		ret = tcpx_setup_socket(pep->sock);
-		if (ret) {
-			goto err;
-		}
 		ret = bind(pep->sock, pep->info->src_addr,
-		      (socklen_t) pep->info->src_addrlen);
+			  (socklen_t) pep->info->src_addrlen);
 	} else {
-		ret = tcpx_setup_socket_nodelay(pep->sock);
-		if (ret) {
-			goto err;
-		}
```

Comment:
This is reverting a change made by this commit:
https://github.com/ofiwg/libfabric/commit/ae62148c8d73b8bdd001f15c87e97ca7583720c9
tcpx_bind_to_port_range() searches for an available port to use.  Why do we want to set that port to re-usable?